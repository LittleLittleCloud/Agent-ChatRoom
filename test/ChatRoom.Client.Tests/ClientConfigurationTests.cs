using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json;
using System.Threading.Tasks;
using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using ChatRoom.SDK;
using FluentAssertions;
using Json.Schema;
using Json.Schema.Generation;
using Xunit;

namespace ChatRoom.Client.Tests;

public class ClientConfigurationTests
{
    [Fact]
    [UseReporter(typeof(DiffReporter))]
    [UseApprovalSubdirectory("ApprovalTests")]
    public void VerifyConfigurationSchema()
    {
        var schema = new JsonSchemaBuilder()
            .FromType<ChatRoomServerConfiguration>()
            .Build();

        var json = JsonSerializer.Serialize(schema, new JsonSerializerOptions { WriteIndented = true });

        var schemaFileName = "client_configuration_schema.json";
        var schemaFilePath = Path.Join("Schema", schemaFileName);
        var schemaFile = File.ReadAllText(schemaFilePath);

        Approvals.Verify(json);
        schemaFile.Should().BeEquivalentTo(json);
    }
}
