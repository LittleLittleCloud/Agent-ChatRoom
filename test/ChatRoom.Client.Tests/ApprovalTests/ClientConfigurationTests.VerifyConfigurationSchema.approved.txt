{
  "type": "object",
  "properties": {
    "room_config": {
      "type": "object",
      "properties": {
        "room": {
          "$ref": "#/$defs/string"
        },
        "port": {
          "$ref": "#/$defs/integer"
        }
      },
      "description": "The configuration for the chat room"
    },
    "channel_config": {
      "type": "object",
      "properties": {
        "openai_config": {
          "$ref": "#/$defs/openAIClientConfiguration"
        }
      },
      "description": "The configuration for the chat channel"
    },
    "agent_extensions": {
      "type": "array",
      "items": {
        "type": "object",
        "properties": {
          "name": {
            "type": "string",
            "description": "The name of the extension, can be any string"
          },
          "command": {
            "type": "string",
            "description": "The command to start the agent extension"
          }
        }
      }
    },
    "name": {
      "type": "string",
      "description": "Your name in the chat room",
      "default": "User"
    }
  },
  "$defs": {
    "string": {
      "type": "string",
      "description": "The name of the room. Default is \u0027room\u0027"
    },
    "integer": {
      "type": "integer",
      "description": "The port number where the room is hosted. Default is 30000"
    },
    "openAIClientConfiguration": {
      "type": "object",
      "properties": {
        "llm_type": {
          "$ref": "#/$defs/lLMType"
        },
        "openai_api_key": {
          "$ref": "#/$defs/string1"
        },
        "openai_model_id": {
          "$ref": "#/$defs/string2"
        },
        "azure_openai_endpoint": {
          "$ref": "#/$defs/string3"
        },
        "azure_openai_key": {
          "$ref": "#/$defs/string4"
        },
        "azure_openai_deploy_name": {
          "$ref": "#/$defs/string5"
        },
        "third_party_llm_endpoint": {
          "$ref": "#/$defs/string6"
        },
        "third_party_llm_key": {
          "$ref": "#/$defs/string7"
        },
        "third_party_llm_model_id": {
          "$ref": "#/$defs/string8"
        }
      },
      "description": "openai configuration, this will be used to create openai client"
    },
    "lLMType": {
      "enum": [
        "AOAI",
        "OpenAI",
        "ThirdParty"
      ],
      "description": "LLM provider, default is Azure OpenAI"
    },
    "string1": {
      "type": "string",
      "description": "OpenAI API key, default is $env:OPENAI_API_KEY"
    },
    "string2": {
      "type": "string",
      "description": "OpenAI model ID, default is gpt-3.5-turbo"
    },
    "string3": {
      "type": "string",
      "description": "Azure OpenAI endpoint, default is $env:AZURE_OPENAI_ENDPOINT"
    },
    "string4": {
      "type": "string",
      "description": "Azure OpenAI key, default is $env:AZURE_OPENAI_API_KEY"
    },
    "string5": {
      "type": "string",
      "description": "Azure OpenAI deploy name, default is $env:AZURE_OPENAI_DEPLOY_NAME"
    },
    "string6": {
      "type": "string",
      "description": "Third-party LLM provider endpoint, default is $env:THIRD_PARTY_LLM_ENDPOINT"
    },
    "string7": {
      "type": "string",
      "description": "Third-party LLM provider key, default is $env:THIRD_PARTY_LLM_KEY"
    },
    "string8": {
      "type": "string",
      "description": "Third-party LLM provider model ID, default is $env:THIRD_PARTY_LLM_MODEL_ID"
    }
  }
}